vortex-buffer/src/buffer.rs

```diff
@@ -234,6 +254,26 @@ impl<T> Buffer<T> {
         }
     }
 
+    /// Returns a slice of self that is equivalent to the given subset.
+    ///
+    /// When processing the buffer you will often end up with &\[T\] that is a subset
+    /// of the underlying buffer. This function turns the slice into a slice of the buffer
+    /// it has been taken from.
+    ///
+    /// # Panics:
+    /// Requires that the given sub slice is in fact contained within the Bytes buffer; otherwise this function will panic.
+    pub fn slice_ref(&self, subset: &[T]) -> Self {
+        let subset_u8 =
```

Comment:
You're forgetting to check the new alignment. 

See `slice` - you cannot end up with a `Buffer` that _doesn't_ respect the defined alignment so you must also panic if the subset_u8 isn't aligned per the current alignment.